rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -515,7 +498,7 @@
                 <javaclass references="org.hibernate.metamodel.spi.{type}"/>
             </when>
             <perform>
-                <hint title="org.hibernate.metamodel.spi.{type} was moved in Hibernate 5" effort="1" category-id="mandatory">
+                <hint title="Hibernate 5 - Moved method org.hibernate.metamodel.spi.{type}" effort="1" category-id="optional">
```

Comment:
@mareknovotny switching to another version of hibernate/JPA is definitely optional while moving to EAP 6/7.

There is no real direct knowledge article explaining that. Here is what I tend to recommend when I do a review:

RATIONALE
------------
One or several object-relational mapping (ORM) frameworks have been detected within the application (eclipselink, eclipselink-jpa, hibernate, hibernate-jpa, lift-jpa, scala-jpa, spring-data-jpa, ...) These frameworks may deviate from the out-of-the-box Java EE supported ORM in JBoss EAP.

RECOMMENDATION
---------------------
All used ORM frameworks should be listed and analyzed more thoroughly to prevent conflicts.

No update to the latest Hibernate or JPA version is required. The currently used ORM frameworks can remain embedded within the application while disabling the EAP out-of-the-box ORM using explicit class loading dependencies (jboss-deployment-structure.xml).
At the same time, an enterprise architecture decision might be taken to update or replace all ORM frameworks by the latest JPA standard. Only the version of Hibernate shipped with the EAP release is fully supported.

REFERENCES
--------------
https://access.redhat.com/solutions/306493
https://access.redhat.com/solutions/707243